main.js

```diff
@@ -1,173 +1,227 @@
-function getJIRAFeed(callback, errorCallback){
-    var user = document.getElementById("user").value;
-    if(user == undefined) return;
-    
-    var url = "https://jira.secondlife.com/activity?maxResults=50&streams=user+IS+"+user+"&providers=issues";
-    make_request(url, "").then(function(response) {
-      // empty response type allows the request.responseXML property to be returned in the makeRequest call
-      callback(url, response);
-    }, errorCallback);
-}
 /**
  * @param {string} searchTerm - Search term for JIRA Query.
- * @param {function(string)} callback - Called when the query results have been  
+ * @param {function(string)} callback - Called when the query results have been
  *   formatted for rendering.
  * @param {function(string)} errorCallback - Called when the query or call fails.
  */
-async function getQueryResults(s, callback, errorCallback) {                                                 
-    try {
-      var response = await make_request(s, "json");
-      callback(createHTMLElementResult(response));
-    } catch (error) {
-      errorCallback(error);
-    }
-}
 
-function make_request(url, responseType) {
-  return new Promise(function(resolve, reject) {
+// Network Request & Utility Functions
+
+const make_request = (url, responseType) => {
+  return new Promise((resolve, reject) => {
     var req = new XMLHttpRequest();
-    req.open('GET', url);
+    req.open("GET", url);
     req.responseType = responseType;
 
-    req.onload = function() {
+    req.onload = () => {
       var response = responseType ? req.response : req.responseXML;
-      if(response && response.errorMessages && response.errorMessages.length > 0){
+      if (
+        response &&
+        response.errorMessages &&
+        response.errorMessages.length > 0
+      ) {
         reject(response.errorMessages[0]);
         return;
       }
       resolve(response);
     };
 
-    // Handle network errors
-    req.onerror = function() {
-      reject(Error("Network Error"));
-    }
-    req.onreadystatechange = function() { 
-      if(req.readyState == 4 && req.status == 401) { 
-          reject("You must be logged in to JIRA to see this project.");
-      }
-    }
-
-    // Make the request
+    handleNetworkError(req);
```

Comment:
naming - by calling this a handler I expect it to handle errors, not wire up functions.